src/charm_state.py

```diff
@@ -94,6 +80,7 @@ class SynapseConfig(BaseModel):  # pylint: disable=too-few-public-methods
     enable_mjolnir: bool = False
     enable_password_config: bool = True
     federation_domain_whitelist: str | None = Field(None)
+    ip_range_whitelist: str | None = Field(None)
```

Comment:
Should we do some more validation here? I noticed that later we're splitting it by comma, stripping whitespace and I'm guessing relying on some formatting of IP addresses?

Reply:
Instead of re-creating the validation that Synapse does, I found out that there is a script that validates the entire configuration file that will be called by the method validate_config. If there is something wrong, the charm will be blocked so the user can review the configuration.

Reply:
Does it say specifically what is wrong? There also seem to be charm specific things like that it is a comma-separated list, right?

Reply:
Yes, it does. Here is an example of the output that will be available in the log: `('Invalid IP range provided: banana.', ('ip_range_whitelist',))`